sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/diagnostics/SparkTaskContext.scala

```diff
@@ -22,22 +25,26 @@ private[spark] case class DeleteOperation(sparkTaskContext: SparkTaskContext, it
 private[spark] case class ReplaceOperation(sparkTaskContext: SparkTaskContext, itemIdentifier: CosmosItemIdentifier)
   extends WriteOperation
 
-private[spark] case class SparkTaskContext(correlationActivityId: String,
+private[spark] case class SparkTaskContext(correlationActivityId: UUID,
                                            stageId: Int,
                                            partitionId: Long,
                                            taskAttemptId: Long,
                                            details: String) extends OperationContext {
 
+  requireNotNull(correlationActivityId, "cosmosClientConfig")
```

Comment:
should the parameter name be "correlationActivityId" instead of "cosmosClientConfig"?

Reply:
:-) yes - fixed in next iteration